sel/src/Sel/Hashing/Generic.hs

```diff
@@ -64,6 +66,24 @@ import LibSodium.Bindings.GenericHashing (cryptoGenericHash, cryptoGenericHashBy
 -- @since 0.0.1.0
 newtype HashKey = HashKey (ForeignPtr CUChar)
 
+instance Eq HashKey where
+ (HashKey pk1) == (HashKey pk2) = 
+    let p = unsafeForeignPtrToPtr pk1
+        q = unsafeForeignPtrToPtr pk2
```

Comment:
This is probably not what you want. You want to use `withForeignPointer` to get at the `Ptr` inside, then compare as before.